app/include/zmk/events/layer_state_changed.h

```diff
@@ -8,17 +8,18 @@
 
 #include <zephyr.h>
 #include <zmk/event_manager.h>
+#include <zmk/keymap.h>
 
 struct zmk_layer_state_changed {
-    uint8_t layer;
-    bool state;
+    zmk_keymap_layers_state_t prior_state;
```

Comment:
We should also have `new_state` here if we're keeping a prior state.

Reply:
That was my original plan but I couldn't find a compelling reason to include it since it's already available.  If you like it that's compelling enough.  I'll add it.